newIDE/app/src/InstancesEditor/InstancesList/index.js

```diff
@@ -105,20 +122,86 @@ export default class InstancesList extends Component<Props, State> {
     }
   };
 
+  _renderLockCell = ({ rowData }: { rowData: RenderedRowInfo }) => {
+    return (
+      <IconButton
+        size="small"
+        onClick={() => {
+          rowData.instance.setLocked(!rowData.instance.isLocked());
+        }}
+      >
+        {rowData.instance.isLocked() && <Lock />}
+        {!rowData.instance.isLocked() && <LockOpen />}
+      </IconButton>
+    );
+  };
+
   _selectFirstInstance = () => {
     if (this.renderedRows.length) {
-      this.props.onSelectInstances([this.renderedRows[0].instance]);
+      this.props.onSelectInstances([this.renderedRows[0].instance], false);
     }
   };
 
+  _sort = ({
+    sortBy,
+    sortDirection,
+  }: {
+    sortBy: string,
+    sortDirection: SortDirection,
+  }) => {
+    this.setState({ sortBy, sortDirection });
+  };
+
+  _orderRenderedRows = () => {
+    this.renderedRows.sort(
+      (a: RenderedRowInfo, b: RenderedRowInfo): number => {
+        const direction =
+          this.state.sortDirection === SortDirection.ASC ? 1 : -1;
+        const compString = (x: string, y: string): number => {
+          if (typeof x === 'string' && typeof y === 'string') {
+            x = x.toLowerCase();
+            y = y.toLowerCase();
+          }
+
+          if (x < y) return direction * 1;
+          if (x > y) return direction * -1;
+          return 0;
+        };
```

Comment:
Move this out of the function and out of the component. This will help reducing the pressure of the JS engine garbage collector by avoiding to re-declare a function.
Also use no abbreviations: `compareStrings`.